Update eas-build-archive.md (#76)
expo/fyi: eas-build-archive.md

When you run `eas build`, we need to get your source code from your development machine to one of our macOS or Linux build workers. In order to do this, EAS CLI will collect and compress project files into a single archive, and then upload that archive to private cloud storage that is accessible only to the build worker. By default, EAS CLI will produce the archive by copying all files starting from the root of the git repository with the exception of `.git`, `node_modules`, and all files matched by rules from `.gitignore` (or `.easignore` if it exists).

> > `.easignore` supports the same format as `.gitignore` files, but it can only be located in the root of your git repository and if it is present non of the existing `.gitignore` files will be respected.

Reviewer: ```suggestion
> `.easignore` supports [the same path patterns as `.gitignore` files](http://git-scm.com/docs/gitignore), but it can only be located in the root of your git repository and, if it is present, none of the existing `.gitignore` files will be respected by EAS CLI.
```

If you set `{ "cli": { "requireCommit": true } }` at the root level of your `eas.json`, EAS CLI will use the `git clone --depth 1 ...` command to create a shallow clone of your repository. With this approach, the project uploaded to EAS Build will be exactly the same as in Git - including information like the branch, commit hash, and so on.

> > Note: `.easignore` files are not supported if you set `requireCommit: true` in your eas.json

Reviewer: ```suggestion
> Note: `.easignore` is not supported if you set `requireCommit: true` in your `eas.json`
```

## How do I opt out of using Git?

The default workflow does not require that you commit your changes, but `git` itself is still used to read some metadata about the repository. You can set the `EAS_NO_VCS=1` environment variable to skip using Git for all EAS CLI commands and optionally `EAS_PROJECT_ROOT` to define the root of your project if it is different than location of your `eas.json` file.

> If you want to use `.easignore` file in this mode, you need to place it in the directory pointed by `EAS_PROJECT_ROOT` env if it's set or in directery with `eas.json` otherwise.

Reviewer: ```suggestion
If you want to use `.easignore` instead of `.gitignore` with `EAS_NO_VCS=1`, then you need to place it in the directory pointed to by the `EAS_PROJECT_ROOT` environment variable (if it's set), or in the same directory as your `eas.json` otherwise.
```
